src/hotspot/share/gc/g1/g1CodeCacheRemSet.cpp

```diff
@@ -40,104 +39,41 @@
 G1CodeRootSetTable* volatile G1CodeRootSetTable::_purge_list = NULL;
 
 size_t G1CodeRootSetTable::mem_size() {
-  return sizeof(G1CodeRootSetTable) + (entry_size() * number_of_entries()) + (sizeof(HashtableBucket<mtGC>) * table_size());
-}
-
-G1CodeRootSetTable::Entry* G1CodeRootSetTable::new_entry(nmethod* nm) {
-  unsigned int hash = compute_hash(nm);
-  return (Entry*)Hashtable<nmethod*, mtGC>::new_entry(hash, nm);
-}
-
-void G1CodeRootSetTable::remove_entry(Entry* e, Entry* previous) {
-  int index = hash_to_index(e->hash());
-  assert((e == bucket(index)) == (previous == NULL), "if e is the first entry then previous should be null");
-
-  if (previous == NULL) {
-    set_entry(index, e->next());
-  } else {
-    previous->set_next(e->next());
-  }
-  free_entry(e);
-}
-
-G1CodeRootSetTable::~G1CodeRootSetTable() {
-  for (int index = 0; index < table_size(); ++index) {
-    for (Entry* e = bucket(index); e != NULL; ) {
-      Entry* to_remove = e;
-      // read next before freeing.
-      e = e->next();
-      BasicHashtable<mtGC>::free_entry(to_remove);
-    }
-  }
-  assert(number_of_entries() == 0, "should have removed all entries");
+  return sizeof(*this) +
+    _table.table_size() * sizeof(Table::Node*) +
+    _table.number_of_entries() * sizeof(Table::Node);
```

Comment:
Instead of exporting Table::Node and the details of the implementation, maybe ResourceHashtable should have a function that returns its footprint? I don't know what name we usually use for that. Maybe `ResourceHashtable::mem_size()` to follow the example of `G1CodeRootSetTable::mem_size()`?

Reply:
I am back after vacation, I added a `ResourceHashtable::mem_size()`, does it look okay?